content/fundamentals/circular-dependency.md

```diff
@@ -6,6 +6,8 @@ While circular dependencies should be avoided where possible, you can't always d
 
 We also describe resolving circular dependencies between modules.
 
+> warning **Warning** A circular dependency might also be caused when using "barrel files"/index.ts files to group imports. Barrel files should be omitted when it comes to modules/providers. For more details please also see [this github issue](https://github.com/nestjs/nest/issues/1181#issuecomment-430197191). 
```

Comment:
I think we could re-word this to something along the lines of "Barrel files should not be used when importing files within the same directory as the barrel file, i.e. `cats/cats.controller` should not import `cats` to import the `cats/cats.service` file".

Reply:
@jmcdo29 Thanks for the review. I have updated the text as suggested.